src/ajax.js

```diff
@@ -24,6 +28,10 @@ const GET = 'GET';
 const POST = 'POST';
 const CTYPE = 'Content-Type';
 
+export const processRequestOptions = hook('async', function(options) {
+  return options;
```

Comment:
Putting the activity check in here (instead of in a separate branch) would simplify things a bit IMO. You wouldn't need to remember to check for `hasDeviceAccess` (or any future changes in the logic) in `tcfControl`, it could just flip the flag if consent & module information are missing and let core do the "normal" check on top.